### DISCUSSION TITLE
Wrong picture for Australian galah?

### DISCUSSION THREAD

**Discussion by Alienraider** [mcguirepr89/BirdNET-Pi#700] [General]
Hi there, Brand new install, brand new user - have to say I love it! But I have a small issue - the detection picture my system brings up on the overview page and elsewhere of an Australian galah is clearly not an Australian galah but rather (I think) a rainbow lorikeet. Not sure if this is something I did or something in the system, and clearly it's not a showstopper rather just an annoyance.  Anyway, is it something I did? Is there anything I can do to fix it?

**Comment by lloydbayley** [comment]
G'day. I've had this happen a couple of times. I assume you're pulling the images from Flickr and on the odd occasion, it pulls the wrong image as I have a feeling there might be a few mistakes in Flickr. Whilst I haven't gone into this too deeply, as it hasn't happened much, but it's certainly worth keeping an eye on.
Have you checked the link that is shown when you click on the picture just to see? Also could be a cache issue?

**Reply by Alienraider** [reply]
Thanks for the reply - yep - it does go off to Flickr and somehow consistently produces the wrong image. Again, it's more annoying than awful! Yesterday I also had the image of a little raven replaced by what I think was a bald eagle! But that was temporary, and it's now pulling what looks like the right image, but still it does seem like something that happens a reasonable amount.

Any suggestions as to how I can clear the image cache and get it to find a fresh image?

**Reply by lloydbayley** [reply]
Hmm...did you check the URL it was giving and make sure it was the correct one? That should tell you whether Flickr has the image wrong.

**Reply by Alienraider** [reply]
It's drawing the image from Flickr - my guess is you're right, it's linking to the wrong image (I haven't had a detection today to confirm and can't find anywhere else where I can find the image!)  Is there a way to correct it?

**Reply by ehpersonal38** [reply]
On the navigation bar at the top of your BirdNET-Pi interface, try navigating to "Best Recordings". From there, click the Australian Galah, and scroll down a bit until you see all the images. These are loaded from Flickr, so if most of them look like the incorrect species, we can investigate further. But if they mostly look correct, then I'd chalk this one up to just people who submit the photos on Flickr mislabeling their photos, or the Flickr search algorithm just being dumb sometimes.

**Reply by lloydbayley** [reply]
Yeh, that's what I meant but it was stupid-o-clock in the morning and @ehpersonal38  said it better! :) It's possible some people have uploaded/named the wrong thing to Flickr. This happens when you deal with the public! :)

**Reply by Alienraider** [reply]
It is indeed drawing the wrong picture from Flickr - helpfully (or otherwise) the photographer realised their error but did not correct the title to the image! Is there a way to exclude that image (or get it to download an alternative)? As far as I know the rest of the images are OK (other than for the odd glitch here and there).

**Reply by lloydbayley** [reply]
Hmm, not easily without re-writing some code just for that particular image, which is not really feasible for a variety of reasons. 
You could restrict your Flickr retrieves to one specific user  but that would presume that that user has ALL the birds you need uploaded to their account. Unless @ehpersonal38 has any other input, I'm afraid you'll have to put up with it or see if you can contact the author with the invalid image to see if they will correct it.

**Reply by Alienraider** [reply]
Cheers - just my luck that somehow seem to have found the only Australian that didn't know what a galah looks like! I've tried blocking the user through Flickr, maybe that will fix it.

**Reply by lloydbayley** [reply]
No fun when you're dealing with a bunch of galahs! (#localjoke)

**Reply by matthew73210** [reply]
Ah you missed the famous episode a couple months back when it wasn't just birds that were uploaded from flicker, but scantly clad woman whom happened to have a name not too dissimilar from birds.

**Comment by ehpersonal38** [comment]
aaffa6f5c5ebdd3939a26703b3dd4d958aba9d0e solves this by adding support for blacklisting Flickr images by ID. Right now that picture of the "Galah" is hardcoded as an ID to be blacklisted, but it would be nice as a future feature to have a little "blacklist" button on each Flickr image that shows up in the BirdNET-Pi interface.

**Reply by lloydbayley** [reply]
That's an excellent idea, and if it wasn't 1:41am, I might have a crack at it.
Also, this brings up a decision of sorts for future-proofing things.
Would you say that the blacklist be maintained in textfile format as the exclusions are now or (my preferred way) put them in a table in the db?
This really applies to other stuff too for adds. As far as I can see, ideally, all those 'lists' could be in the db.
Thoughts?

**Reply by ehpersonal38** [reply]
I've just implemented this and it's working as envisioned. 😄 Anyone can access it by updating their system and then clicking a bird image in the interface. It will then open the larger view, and a new button will be visible at the bottom "Never show this image again."

@lloydbayley I think you're right that these lists (disk_check_exclude, include_species_list, etc, etc) should be database tables instead of text files. But, 'tis the way we've been doing it 'round here. So as far as I'm concerned for now they can stay.

**Reply by lloydbayley** [reply]
Right. Understood.

**Comment by Alienraider** [comment]
Great fix - thanks!
